[FLINK-18682][orc][hive] Vector orc reader cannot read Hive 2.0.0 table

lirui-apache opened the pull request:
<!--
*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can set up Azure Pipelines CI to do that following [this guide](https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines#AzurePipelines-Tutorial:SettingupAzurePipelinesforaforkoftheFlinkrepository).

  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message (including the JIRA id)

  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.


**(The sections below can be removed for hotfixes of typos)**
-->

## What is the purpose of the change

Fix the issue that vector orc reader doesn't support Hive 2.0.0. Need to solve two problems:
1. `org.apache.hadoop.hive.ql.exec.vector.TimestampColumnVector` is not available in Hive 2.0.0. `AbstractOrcColumnVector::createFlinkVector` will throw CNFE.
2. Timestamp is stored with `LongColumnVector` in 2.0.0.


## Brief change log

  - Add `OrcLegacyTimestampColumnVector` to get timestamps from `LongColumnVector`.
  - `OrcLegacyTimestampColumnVector` also provides util methods to decide whether new or legacy column vector should be used.
  - Enable orc read test for 2.0.0 in `TableEnvHiveConnectorITCase`.


## Verifying this change

Existing and new test cases.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? NA

JingsongLi commented on flink-formats/flink-orc/src/main/java/org/apache/flink/orc/vector/OrcLegacyTimestampColumnVector.java:
```diff
@@ -0,0 +1,168 @@
+
+	static {
+		try {
+			hiveTSColVectorClz = Class.forName("org.apache.hadoop.hive.ql.exec.vector.TimestampColumnVector");
```
Why use reflection? why not just use `OrcTimestampColumnVector`?

JingsongLi replied:
Can you have a `TimestampUtil`,  and then decide using `OrcTimestampColumnVector` or a `LongOrcTimestampVector`?

lirui-apache commented:
@JingsongLi I have refactored the code and added a util class to decide whether to use new or legacy timestamp vector. Please have another look, thanks.
